pic-sure-auth-services/src/main/java/edu/harvard/hms/dbmi/avillach/auth/service/auth/AuthorizationService.java

```diff
@@ -192,6 +199,11 @@ public boolean isAuthorized(Application application , Object requestBody, UUID u
                 }
             }
 
+            keys.add((accessRule.getCheckMapKeyOnly()==null)?"null":Boolean.toString(accessRule.getCheckMapKeyOnly()));
```

Comment:
Don't like this. This is expensive operation and you are doing it twice! Don't like it! Would looooove to add a ticket to make this more efficient (IN THE FUTURE)